### PR TITLE
Laboratory work #1, Biryukova Ksenia - 19FPL2

### PR DESCRIPTION
**Pull request by biryukovaksenia**

### CODE REVIEW

**Comment by ValeriyaKuznetsova** [comment]
Please, rename your Pull Request. Use this template – Laboratory work #1, Fedorov Makar - 19FPL1

**Comment by ValeriyaKuznetsova** [comment]
Unfortunately, the deadline is not that far. Please, push changes so that we can review them.

File: lab_1/main.py [hunk 1]
```diff
@@ -2,18 +2,35 @@
 Lab 1
 A concordance extraction
 """
+import re
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: remove .DS_Store files, please

File: lab_1/main.py [hunk 2]
```diff
@@ -2,18 +2,35 @@
...
 
+def tokenize(text: str) -> list:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: leave the function description (everywhere)

def tokenize(text) :
     """
     Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
     :param text: the initial text
     :return: a list of lowercased tokens without punctuation
     e.g. text = 'The weather is sunny, the man is happy.'
     --> ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
     """
   YOUR CODE

File: lab_1/main.py [hunk 3]
```diff
@@ -2,18 +2,35 @@
...
+    try:
+        text = text.lower()
+        match = re.findall(r"\w+-\w+", text)
+        match_new = []
+
+        for i in match:
+            new = i.replace('-', '')
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: can you update regex that it doesn't include "-"?

File: lab_1/main.py [hunk 4]
```diff
@@ -2,18 +2,35 @@
...
+            text = text.replace(i, new)
+
+        match2 = re.findall(r"\w+'\w", text)
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: same

**Токены могут состоять только из букв английского алфавита (не допускаются цифры и любые другие символы внутри слова).
При этом любые символы (кроме букв) внутри токена удаляются, например, don't заменяется на dont, round-up на roundup и т.д.**

Comment (ValeriyaKuznetsova) [reply lab_1/main.py]: if it was a way to remove punctuation from a text, consider that there can be "", <>, ?, !, etc.

File: lab_1/main.py [hunk 5]
```diff
@@ -26,7 +43,16 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
     stop_words = ['the', 'is']
     --> ['weather', 'sunny', 'man', 'happy']
     """
-    pass
+    tokens2 = []
+    if type(tokens) == list:
+        if type(stop_words) == list:
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: can you join lines 47 and 48?

File: lab_1/main.py [hunk 6]
```diff
@@ -26,7 +43,16 @@ def remove_stop_words(tokens: list, stop_words: list) -> list:
...
+            for i in stop_words:
+                if i in tokens:
+                    tokens.remove(i)
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: try this:
stop_words = ['the', 'is']
tokens = ['the', 'is', 'a', 'the']

The expected result is ['a']

**Comment by ValeriyaKuznetsova** [comment]
@biryukovaksenia 
previous issues were not corrected, first, correct them, then, I'll review new ones

**Comment by biryukovaksenia** [comment]
> @biryukovaksenia
> previous issues were not corrected, first, correct them, then, I'll review new ones

What exactly? I rewrited the code

File: lab_1/main.py [hunk 7]
```diff
@@ -2,92 +2,87 @@
...
 """
+import re
 
 
 def tokenize(text: str) -> list:
-    """
-    Splits sentences into tokens, converts the tokens into lowercase, removes punctuation
-    :param text: the initial text
-    :return: a list of lowercased tokens without punctuation
-    e.g. text = 'The weather is sunny, the man is happy.'
```
Comment (ValeriyaKuznetsova) [inline lab_1/main.py]: File descriptions should be present, please, change back (everywhere)

**Review by ValeriyaKuznetsova — COMMENTED** [review]
do not forget about start.py
write_to_file should be implemented too

correct some issues in lint

File: lab_1/start.py [hunk 8]
```diff
@@ -2,17 +2,36 @@
...
+
+    adjacent_words = main.get_adjacent_words(tokens, 'people', 5, 5)
+    print('adjacent_words for the word: ', adjacent_words[:3])
+
+    main.write_to_file('report.txt', adjacent_words)
+
+    RESULT = adjacent_words
     # DO NOT REMOVE NEXT LINE - KEEP IT INTENTIONALLY LAST
-    assert RESULT == [(), ()], 'Concordance not working'
+    assert RESULT, 'Concordance not working'
```
Comment (ValeriyaKuznetsova) [inline lab_1/start.py]: RESULT == [expected result]

for example, [['the', 'man']]

Comment (ValeriyaKuznetsova) [reply lab_1/start.py]: correct this 
everything else seems good

**Comment by ValeriyaKuznetsova** [comment]
Open another PR with the second lab

**Pull request closed without merging**